### PR TITLE
Add CAN data to car state

### PR DESCRIPTION
**Pull request by garned**

### CODE REVIEW

File: extras/format.sh [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 # clang formatting
 echo "Start formatting source files.."
+# find . -not -path "./.*" -name "*.cpp" -o -name "*.c" -o -name "*.h" | xargs -I {} clang-format -i {}
+# find . -type d \( -path "./.*" -o -path "./datasheets" -o -path "./medis" \) -prune -false -o -name "*.h" -o -name "*.hpp" -o -name "*.cpp" | xargs clang-format -i {}
```
Comment (andreaskuster) [inline extras/format.sh:5]: I assume this is a leftover?

File: lib/CAN/CANBus.cpp [hunk 2]
```diff
@@ -10,16 +10,106 @@
...
+  this->max_ages[BMS_BASE_ADDR | 0x2] = MAXAGE_CMU_VOLTAGES; // CMU1
+  this->max_ages[BMS_BASE_ADDR | 0x3] = MAXAGE_CMU_VOLTAGES; // CMU1
+  this->max_ages[BMS_BASE_ADDR | 0x5] = MAXAGE_CMU_VOLTAGES; // CMU2
+  this->max_ages[BMS_BASE_ADDR | 0x6] = MAXAGE_CMU_VOLTAGES; // CMU2
+  this->max_ages[BMS_BASE_ADDR | 0x8] = MAXAGE_CMU_VOLTAGES; // CMU3
+  this->max_ages[BMS_BASE_ADDR | 0x9] = MAXAGE_CMU_VOLTAGES; // CMU3
+  this->max_ages[BMS_BASE_ADDR | 0xF4] = MAXAGE_PACK_SOC;
+  this->max_ages[BMS_BASE_ADDR | 0xF5] = MAXAGE_BALANCE_SOC;
+  this->max_ages[BMS_BASE_ADDR | 0xF6] = MAXAGE_CHARGER_CONTROL;
+  this->max_ages[BMS_BASE_ADDR | 0xF7] = MAXAGE_PRECHARGE_STATUS;
```
Comment (andreaskuster) [inline lib/CAN/CANBus.cpp:35]: In modern C++ you should be able to omit the `this->` as long as max_ages as long as the variable is defined in the class. Please do so.

File: lib/CAN/CANBus.cpp [hunk 3]
```diff
@@ -117,20 +237,29 @@ void CANBus::task() {
...
+        carState.Mppt3Current = packet.getData_f32(1);
+        // carState.PhotoVoltaicCurrent = carState.Mppt1Current + carState.Mppt2Current + carState.Mppt3Current;
+
         // MPPT3 Output Voltage V packet.getData_f32(0)
         // MPPT3 Output Current A packet.getData_f32(1)
         break;
       }
     }
-    this->sleep(50);
+    this->sleep(CAN_TASK_WAIT);
```
Comment (andreaskuster) [inline lib/CAN/CANBus.cpp:262]: same

**Review by andreaskuster — COMMENTED** [review]
Generally looks good to me. 

Only the this-> legacy code should be removed.

File: extras/format.sh (hunk 1 above)
Comment (garned) [reply extras/format.sh:5]: The original command doens't work on my machine. The one on line 5 does though.

Comment (andreaskuster) [reply extras/format.sh:5]: Ok strange. What version are you using (clang-format) and what is the error message?

Comment (garned) [reply extras/format.sh:5]: 10.0.0-4ubuntu1

**Review by andreaskuster — APPROVED** [review]
LGTM!

**Pull request merged**